drivers/serial/uart_mcux_flexcomm.c

```diff
@@ -346,6 +378,577 @@ static int mcux_flexcomm_uart_config_get(const struct device *dev,
 }
 #endif /* CONFIG_UART_USE_RUNTIME_CONFIGURE */
 
+#ifdef CONFIG_UART_ASYNC_API
+/* This function is called by this driver to notify user callback of events */
+static void async_user_callback(const struct device *dev,
+					      struct uart_event *evt)
+{
+	const struct mcux_flexcomm_data *data = dev->data;
+
+	if (data->async_callback) {
+		data->async_callback(dev, evt, data->async_cb_data);
+	}
+}
+
+static int mcux_flexcomm_uart_callback_set(const struct device *dev,
+					   uart_callback_t callback,
+					   void *user_data)
+{
+	struct mcux_flexcomm_data *data = dev->data;
+
+	data->async_callback = callback;
+	data->async_cb_data = user_data;
+
+	return 0;
+}
+
+static int mcux_flexcomm_uart_tx(const struct device *dev, const uint8_t *buf,
+				 size_t len, int32_t timeout)
+{
+	const struct mcux_flexcomm_config *config = dev->config;
+	struct mcux_flexcomm_data *data = dev->data;
+	int ret = 0;
+
+	if (config->tx_dma.dev == NULL) {
+		return -ENODEV;
+	}
+
+	unsigned int key = irq_lock();
+
+	/* Getting DMA status to tell if channel is busy or not set up */
+	struct dma_status dma_status;
```

Comment:
I guess the Zephyr coding guidelines require the name of the variable to be different from the structure name.

Reply:
it seems to be okay with just `status`